Support for SRILM maxent LMs

wintrode opened the pull request:
Added support to download and build the libLBFGS library, required by the SRILM toolkit for maxent LMs, and modfied train_lms_srilm.sh to build maxent LMs and store in ARPA format.

danpovey committed:
minor fix to cu-kernels.cu (use relative pathname); efficiency and clarity improvements to some nnet compilation code.

freewym committed:
Add an option to discard any partial minibatches of uneven size that are encountered at the end in nnet3-merge-egs

jtrmal committed:
mainly fixing the phoneme confusion matrix generation for proxy KWS

hainan-xv committed:
Fixed a bug in fstdeterminizestar which might produce wrong weights under log-semiring. Did some stylistic changes

Commit:
Document the conventional use of WIP: prefix for work-in-progress pull requests

danpovey committed:
adding more testing code for various core utilities, including lattice-align-words.

danpovey committed:
some fixes to the previous commit

danpovey committed:
fix to compiler-specific compilation problem

freewym committed:
starting for blstm

wintrode committed:
Adding support for building SRILM with libLBFGS and maxent LM support

jtrmal commented:
Jonathan, thanks -- however the PR includes lots of changes from elsewhere
-- would you mind figuring out what is wrong?
I guess rebase might help.
you could also cherry-pick the two commits into separate branch and create
PR from this separate branch.
y.

On Tue, Nov 10, 2015 at 10:43 PM, wintrode notifications@github.com wrote:

> Added support to download and build the libLBFGS library, required by the
> SRILM toolkit for maxent LMs, and modfied train_lms_srilm.sh to build
> 
> ## maxent LMs and store in ARPA format.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/jtrmal/kaldi/pull/1
> Commit Summary
> - minor fix to cu-kernels.cu (use relative pathname); efficiency and
>   clarity improvements to some nnet compilation code.
> - mainly fixing the phoneme confusion matrix generation for proxy KWS
> - Merge pull request #285 from jtrmal/babelfix
> - Fixed a bug in fstdeterminizestar which might produce wrong weights
>   under log-semiring. Did some stylistic changes
> - Add an option to discard any partial minibatches of uneven size that
>   are encountered at the end in nnet3-merge-egs
> - Merge pull request #284 from freewym/fix
> - Merge pull request #286 from hainan-xv/determinizestarcommit
> - Document the conventional use of WIP: prefix for work-in-progress
>   pull requests
> - Merge pull request #287 from kkm000/document-wip
> - adding more testing code for various core utilities, including
>   lattice-align-words.
> - some fixes to the previous commit
> - fix to compiler-specific compilation problem
> - Merge pull request #288 from danpovey/more-testing
> - Adding support for building SRILM with libLBFGS and maxent LM support
> - Merging support for SRILM maxent LMs with remote master
> 
> File Changes
> - _M_ egs/babel/s5c/conf/lang/107-vietnamese-fullLP.official.conf
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-0 (2)
> - _M_ egs/babel/s5c/local/datasets/extra_kws.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-1 (2)
> - _M_ egs/babel/s5c/local/generate_confusion_matrix.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-2 (26)
> - _M_ egs/babel/s5c/run-4-anydecode.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-3 (12)
> - _M_ egs/scale-pashto/s5/local/train_lms_srilm.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-4 (26)
> - _M_ egs/wsj/s5/local/nnet3/run_tdnn.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-5 (1)
> - _M_ egs/wsj/s5/steps/nnet3/lstm/train.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-6 (4)
> - _M_ egs/wsj/s5/steps/nnet3/train_tdnn.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-7 (3)
> - _M_ src/cudamatrix/cu-kernels.cu
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-8 (2)
> - _M_ src/doc/tutorial_git.dox
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-9 (9)
> - _M_ src/fstext/determinize-star-inl.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-10 (963)
> - _M_ src/fstext/determinize-star-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-11 (3)
> - _M_ src/fstext/determinize-star.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-12 (9)
> - _M_ src/fstext/remove-eps-local-inl.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-13 (4)
> - _M_ src/hmm/Makefile
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-14 (5)
> - _A_ src/hmm/hmm-test-utils.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-15 (274)
> - _A_ src/hmm/hmm-test-utils.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-16 (81)
> - _M_ src/hmm/hmm-topology-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-17 (14)
> - _M_ src/hmm/hmm-topology.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-18 (44)
> - _M_ src/hmm/hmm-topology.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-19 (15)
> - _M_ src/hmm/hmm-utils-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-20 (54)
> - _M_ src/hmm/transition-model-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-21 (36)
> - _M_ src/lat/Makefile
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-22 (2)
> - _A_ src/lat/word-align-lattice-lexicon-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-23 (237)
> - _M_ src/lat/word-align-lattice-lexicon.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-24 (158)
> - _M_ src/lat/word-align-lattice-lexicon.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-25 (20)
> - _M_ src/latbin/lattice-to-nbest.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-26 (20)
> - _M_ src/nnet2/am-nnet-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-27 (12)
> - _M_ src/nnet2/online-nnet2-decodable-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-28 (15)
> - _M_ src/nnet2bin/nnet-shuffle-egs.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-29 (6)
> - _M_ src/nnet3/nnet-compile-utils-test.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-30 (51)
> - _M_ src/nnet3/nnet-compile-utils.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-31 (160)
> - _M_ src/nnet3/nnet-compile-utils.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-32 (36)
> - _M_ src/nnet3/nnet-computation.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-33 (3)
> - _M_ src/nnet3/nnet-simple-component.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-34 (38)
> - _M_ src/nnet3bin/nnet3-compute.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-35 (29)
> - _M_ src/nnet3bin/nnet3-merge-egs.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-36 (11)
> - _M_ src/nnet3bin/nnet3-shuffle-egs.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-37 (22)
> - _M_ src/tree/context-dep.cc
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-38 (7)
> - _M_ src/tree/context-dep.h
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-39 (4)
> - _A_ tools/extras/install_liblbfgs.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-40 (29)
> - _M_ tools/extras/install_srilm.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-41 (19)
> - _A_ tools/install_liblbfgs.sh
>   https://github.com/jtrmal/kaldi/pull/1/files#diff-42 (1)
> 
> Patch Links:
> - https://github.com/jtrmal/kaldi/pull/1.patch
> - https://github.com/jtrmal/kaldi/pull/1.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jtrmal/kaldi/pull/1.

vijayaditya committed:
Fixed data prep in SWBD, to ensure CV data does not overlap with train in speed perturbed case

wintrode committed:
Revert "fix to compiler-specific compilation problem"

This reverts commit a5094d08feae8ec9a78789a33597dce3bd23d990.

wintrode committed:
Merge branch 'master' of https://github.com/wintrode/kaldi

wintrode committed:
Installation and use of SRILM maxent LMs

wintrode committed:
Merge branch 'master' into scale-pashto-clean

tomkocse committed:
Added sox version checking in ASpIRE recipe, usage message in wav-reverberate.cc and bug fix in local/multi_condition/prepare_impulses_noises.sh.

danpovey committed:
Merge pull request #325 from tomkocse/soxcheck

Added sox version checking in ASpIRE recipe, usage message in wav-rev…

danpovey committed:
various minor changes to get rid of warnings on recent llvm.

mdoulaty committed:
Refactoring run_bnf.sh

added few variables to hold some common paths to avoid repeating them

danpovey committed:
Merge pull request #327 from mrtdoulaty/patch-1

Refactoring run_bnf.sh

dogancan committed:
Fix clang compiler errors.

It seems these were inadvertently introduced while making the code in util/ comply with the Google style (#315).

danpovey committed:
Merge pull request #329 from dogancan/fix-clang-errors

Fix clang compiler errors.

dogancan committed:
Cosmetic fix for the last commit regarding clang compiler errors.

danpovey committed:
Merge pull request #330 from dogancan/fix-clang-errors

Cosmetic fix for the last commit regarding clang compiler errors.

danpovey committed:
Merge pull request #323 from vijayaditya/bug_fix

Fixed data prep in SWBD, to ensure CV data does not overlap with trai…

vijayaditya committed:
removed trivial bug in tedlium recipe, introduced due to previous commit. Doesn't effect anything

danpovey committed:
Merge pull request #333 from vijayaditya/bug_fix

removed trivial bug in tedlium recipe, introduced due to previous com…

Commit:
Fix compilation issues with gcc4.4.7

jtrmal committed:
Merge pull request #335 from speechmatics/test_old_gcc

Fix compilation issues with gcc4.4.7

Commit:
Fixes to unit tests under Windows

danpovey committed:
Merge pull request #340 from kkm000/test-compile-windows

Fixes to unit tests under Windows

naxingyu committed:
fix a convnet script bug

danpovey committed:
Merge pull request #346 from naxingyu/fix-train-convnet-bug

fix a convnet script bug

danpovey committed:
minor fix: Adding onlinebin to voxforge path.sh

danpovey committed:
some partial changes regarding decodable objects

danpovey committed:
voxforge recipe: fix to path.sh, adding onlinebin/.

zhichaowang committed:
modify add elements kernel

jfainberg committed:
num_targets check and message dealt with feat_dim instead of num_targets

bhigy committed:
Fixing link in tutorial "Running the example scripts"

The link is prefixed by the page address unless we prefix it with "https://"

danpovey committed:
Merge pull request #350 from bhigy/tutofix

Fixing link in tutorial "Running the example scripts"

danpovey committed:
Merge pull request #349 from jfainberg/nnet3_changes

make_tdnn_configs.py: num_targets check and error message dealt with feat_dim instead of num_targets

danpovey committed:
very minor fix to documentation

danpovey committed:
refactoring nnet3 decodable code

zhichaowang committed:
modify add elements kernel

danpovey committed:
Merge pull request #348 from zhichaowang/modify-add-elements-kernel

modify add elements kernel

danpovey committed:
Merge pull request #352 from danpovey/refactor-nnet3-decodable

Refactor nnet3 decodable

danpovey committed:
nnet3: modifying nnet3-compute.cc regarding default acwt (fix regarding previous commit).

david-ryan-snyder committed:
Adding bn_music_speech example which demonstrates music/speech discrimination using the MUSAN corpus and Broadcast news. Also adding binaries compute-vad-from-frame-likes and merge-vads that will be used for frame-level GMM-based VAD.

david-ryan-snyder committed:
music-speech: Cosmetic fix

danpovey committed:
fix bug in nnet3/nnet-am-decodable-simple.cc introduced by recent commit (would cause crash in LSTM decoding).  Thanks to Yiming Wang for noticing

bhigy committed:
Tutorials minor errors (issue #351)

vijayaditya committed:
minor fix to utils/fix_data_dir.sh

jtrmal committed:
Merge pull request #357 from vijayaditya/bug_fix

minor fix to utils/fix_data_dir.sh

danpovey committed:
Merge pull request #356 from bhigy/tutofix

 Tutorials minor errors (issue #351)

danpovey committed:
nnet3: modifying the initialization procedure of the natural-gradient code for efficiency.

david-ryan-snyder committed:
music-speech: Adding sid/compute_vad_decision_gmm.sh, a script that uses GMM likelihoods to compute VAD decisions

david-ryan-snyder committed:
music-speech: minor fix to run.sh

danpovey committed:
A number of small unrelated cosmetic changes.

danpovey committed:
Merge pull request #313 from rikrd/gst-plugin-fix-build-macosx

fix the gst-plugin build script to work with clang

danpovey committed:
improve-natural-gradient-init: improving the robustness of the reorthogonalization process.

danpovey committed:
changing initialization in online-natural-gradient code to avoid an undesirable symmetry.

Commit:
modify arpa-reading code. generate error when n-grams speficied in header and statistics not match

danpovey committed:
Merge pull request #365 from Minhua722/arpa-read

modify arpa-reading code to die if ngram-orders are missing

danpovey committed:
nnet3 training script: cosmetic change to clarify usage

danpovey committed:
remove deprecated append-feats program (paste-feats should be used instead).

chenguoguo committed:
WIP: documented ConstArpaLm stuff in const-arpa-lm.h; fixed I/O bug (was using 25% more of what's needed on the disk); changed lm_states_size_ to int64

hainan-xv committed:
fix a bug where utterance_id is missing for Mikolov's rnnlm rescoring; fix a bug in rnnlmrescore.sh where old lmscore not getting removed under certain conditions

danpovey committed:
Merge pull request #369 from hainan-xv/rnnlm_bugfix

fix a bug where utterance_id is missing for Mikolov's rnnlm rescoring…

Commit:
The partial lattice after lattice-align-words-lexicon wasn't topologically sorted.

alumae committed:
nnet1 to nnet2 conversion: removed an assert that checks whether input frames are contiguous (a remnant from earlier limitation)

danpovey committed:
Merge pull request #371 from alumae/nnet1-to-nnet2-conversion-update

fix to nnet1-to-raw-nnet to allow non-contiguous input frames

danpovey committed:
Merge pull request #370 from speechmatics/align_topo_sort

Partial lattice from lattice-align-words-lexicon not sorted.

xiaohui-zhang committed:
updated nnet2 natural gradient codes according to nnet3 natural gradient codes.

Commit:
Added LM rescoring scripts for CHiME3

danpovey committed:
Merge pull request #372 from xiaohui-zhang/1130

updated nnet3 and nnet3 natural gradient codes (improved initialization,.etc)

danpovey committed:
Merge pull request #312 from freewym/blstm

Merging changes for BLSTM (mostly scripts)

KarelVesely84 committed:
nnet1: minor script updates,

- extract-segments.cc, cosmetic change,

KarelVesely84 committed:
nnet1: adding occassionnal gradient logging to sequence-discriminative training

freewym committed:
some fixes for BLSTM

danpovey committed:
Merge pull request #373 from freewym/blstm

some fixes for BLSTM

chenguoguo committed:
Moved ReadIntegerArray() and WriteIntegerArray() to const-arpa-lm.cc; changed ReadInternalDeprecated() to ReadInternalOldFormat()

chenguoguo committed:
Removed ReadIntegerArray() and WriteIntegerArray() from lm/const-arpa-lm.cc

Commit:
Modified run.sh to check if SRILM and RNNLM are installed at the beginning

danpovey committed:
Merge pull request #367 from chenguoguo/carpa

WIP: documented ConstArpaLm stuff in const-arpa-lm.h; fixed I/O bug (…

Commit:
Removed unneeded and flawed is_sil test from make_lexicon_fst.pl

danpovey committed:
Fix bug in cudamatrix/cu-device.cc whereby if it failed to get a GPU it would print a non-informative error message.

david-ryan-snyder committed:
music-speech: Adding config file needed for bn example

Commit:
Added a variable definition test to prevent warnings

danpovey committed:
Merge pull request #375 from kkm000/is_sil-fix

Fixed index reference in silence test in make_lexicon_fst.pl

tomkocse committed:
Adding the data combining script and program.

tomkocse committed:
Adding the data combining script and program.

tomkocse committed:
Fix issues in concat-feats.cc.

danpovey committed:
various unrelated cosmetic changes

danpovey committed:
fix memory leak in FST-reading code (thanks to Xiang Li for finding the issue)

sw005320 committed:
Added RESULTS

sw005320 committed:
Modified RESULTS usage

sw005320 committed:
Merge remote-tracking branch 'origin/master' into chime3-egs

sw005320 committed:
Update README.txt

sw005320 committed:
Modified comments from "enhan" to "enhanced" etc.

david-ryan-snyder committed:
music-speech: Improving documentation, style, and config handling in binaries compute-vad-from-frame-likes and merge-vads. In egs/bn_music_speech: removing old code from cmd.sh, replacing conf/merge_vad.conf with conf/merge_vad_map.txt (and adding comments to it), changing how python scripts are invoked.

david-ryan-snyder committed:
music-speech: Fixing language in src/ivectorbin/merge-vads.cc

david-ryan-snyder committed:
music-speech: Fixing error messaging (and handling) and documentation in compute-vad-from-frame-likes and merge-vads

danpovey committed:
Merge pull request #354 from david-ryan-snyder/music-speech

WIP Music/Speech Discrimination and GMM-based VAD

KarelVesely84 committed:
swbd: initial commit of confidence calibration script

danpovey committed:
Merge pull request #379 from sw005320/chime3-egs

Chime3 egs

vdp committed:
Fix a lattice determinization bug.
Leads to slight(~3%) running time improvement for "lattice faster" decoding(determinization
is not the most costly operation there), and more noticeable(~30%) lattice size reduction.

danpovey committed:
Merge pull request #382 from vdp/latdet-fix

Fix a lattice determinization bug.

tomkocse committed:
Rewriting steps/cleanup/combine_short_segments.sh in perl. This solves the speed problem.

KarelVesely84 committed:
ami: adding the per-word confidence calibration example (logistic regression),

KarelVesely84 committed:
calibration: debugging previous commit

KarelVesely84 committed:
confidence: reshaping the setup (all file created in one directory)

Commit:
throw GPU usage information (nvidia-smi) when no more unused GPUS

danpovey committed:
Merge pull request #386 from Minhua722/gpu-check

throw GPU usage information (nvidia-smi) when no more unused GPUS

vijayaditya committed:
Modified nnet3-combine to accept single models

danpovey committed:
Merge pull request #387 from vijayaditya/bug_fix

Modified nnet3-combine to accept single models

tomkocse committed:
Change combine_short_segments.sh to combine with contiguous segments.

tomkocse committed:
Change combine_short_segments.sh to work in length in seconds instead of length in frames.

danpovey committed:
Merge pull request #377 from tomkocse/concat_feats

Adding the data combining script and program.

vdp committed:
Small fix to lattice determinization

The EpsilonClosure() method of LatticeDeterminizerPruned and
LatticeDeterminizer classes now sorts the subset vectors by state ID.
This leads to a modest(few percent) lattice size reduction.

david-ryan-snyder committed:
music-speech: Fixing bug in binary merge-vads and in local/make_musan.py for egs/bn_music_speech

david-ryan-snyder committed:
music-speech: Reformatting fix in previous commit so that it doesn't go over the 80 character limit

danpovey committed:
Merge pull request #389 from david-ryan-snyder/music-speech

Bug fixes in music-speech example and binaries

vijayaditya committed:
Added BLSTM results for AMI

jtrmal committed:
Merge pull request #390 from vijayaditya/bug_fix

Added BLSTM results for AMI

danpovey committed:
Merge pull request #388 from vdp/latdet-fix2

Small fix to lattice determinization

danpovey committed:
Improve debug information printed out when we fail to get GPUs.  Remove unused variable in nnet3/nnet-compute.h

tomkocse committed:
Fix the issue when speaker has no enough segments to reach the specified length.

danpovey committed:
Merge pull request #391 from tomkocse/concat_feats

Fix the issue when speaker has no enough segments to reach the specified length

danpovey committed:
changes to make concat-feats less verbose when used with combine_short_segments.sh.

tomkocse committed:
Minor bug fix for end token of component <NaturalGradientAffineComponent>.

Commit:
intial checkin of language pack and decoding scripts

Commit:
intial checkin of language pack and decoding scripts

wintrode committed:
intial check-ins of language pack and decoding scripts

wintrode committed:
intial check-ins of language pack and decoding scripts

wintrode committed:
intial check-ins of language pack and decoding scripts

danpovey committed:
Merge pull request #392 from tomkocse/endtokenfix

Minor bug fix for end token of component <NaturalGradientAffineComponent>.

KarelVesely84 committed:
confidences: removing dependency on 'sclite' for confidence calibration

KarelVesely84 committed:
confidence: improving comments in code, adding tedlium example

vijayaditya committed:
Modified Beamformit URL and install directory name. Recipes using Beamformit should use the new path

danpovey committed:
Merge pull request #394 from vijayaditya/bug_fix

Modified Beamformit URL and install directory name. Recipes using Bea…

chenguoguo committed:
WARNING: this commit may change decoding results slightly due to different pruning effects. Added fstpushspecial to LG stage of graph building to make LG.fst more stochastic. Tested on WSJ, it increased the graph building time slightly, but reduced decoding time. Impact to WER was very small. Also tested that adding fstpushspecial to the HCLG stage was not helpfuf.

danpovey committed:
Merge pull request #401 from chenguoguo/graph

WARNING: this commit may change decoding results slightly due to diff…

tomkocse committed:
Added an option to the nnet3 configs so that the final log softmax layer is selective.

danpovey committed:
Merge pull request #402 from tomkocse/configoption

Added an option to the nnet3 configs so that the final log softmax layer is selective.

david-ryan-snyder committed:
music-speech: Adding symbolic link to the sid directory in egs/bn_music_speech/v1

jtrmal committed:
Merge pull request #403 from david-ryan-snyder/music-speech

music-speech: Adding symbolic link to the sid directory in egs/bn_music_speech/v1

psmit committed:
Make the cstr scripts for wsj use dict suffix

The scripts for the cstr style wsj directory layout did not take in
account the dict_suffix. The script is now brought up-to-date with
wsj_extend_dict.sh

danpovey committed:
Merge pull request #404 from phsmit/fix_cstr_wsj_datacollection

Make the cstr scripts for wsj use dict suffix

KarelVesely84 committed:
beamformit: updating the build script,

- fixing the case when 'libsnd' installed to local dirs

KarelVesely84 committed:
beamformit: removing the build patch, it is no longer needed,

vdp committed:
Minor cosmetic changes in lattice determinization code

Small unrelated fix to src/decoder/Makefile:
Remove the "-O3" compiler flag from that file, because we expect to be
able to set such options globally in "kaldi.mk"

vijayaditya committed:
Added AMI recipe for using clean alignments, minor bug correction in IHM recipe

KarelVesely84 committed:
confidence: implementing Dan's suggestions

- implemented categorical word-identities on inputs, with rare words
  grouped into single class (word-count in dev set < 10)
- removed '--grep-pattern'
- putting ARPA unigram parsing for into extarnal script, called by
  master script
- also tried the neighbor-word confidences as inputs, but no improvement
- partially removing numpy

KarelVesely84 committed:
confidence: adding accidentally missing files

KarelVesely84 committed:
confidence: removing numpy include

danpovey committed:
Merge pull request #405 from vdp/epsclosure-prettify

Minor cosmetic changes in lattice determinization code

danpovey committed:
Merge pull request #406 from vijayaditya/bug_fix

WIP : Added AMI recipe for using clean alignments, minor bug correction in …

vijayaditya committed:
Added results for AMI-SDM nnet3 training with SAT alignments

danpovey committed:
Merge pull request #407 from vijayaditya/bug_fix

Added results for AMI-SDM nnet3 training with SAT alignments

vijayaditya committed:
Added AMI-MDM8 BLSTM results

danpovey committed:
Merge pull request #408 from vijayaditya/bug_fix

Added AMI-MDM8 BLSTM results

KarelVesely84 committed:
confidence: adopting next round of Dans remarks,

- removing dependence on numpy,
- moving word filtering to local/ (using 1/0 flag file)
- moving word-length calculation to local/
- moving lattice_depth_per_frame.sh to steps/conf/ (yes, I intend to use it for other confidence meausres)

- the format 'word-identifier1 feature1 feature2 .. featureN'
  was not adopted at this moment. Instead there is 'words.txt'
  augmented  by 1 column, 1 file per feature.
  The features are 4, the filtering flag, word-lenght, word-`category'
  and unigrams (unigrams have 2 columns only).
  Like this we don't `hard-code' the layout, which could be dangerous
  in case of further extensions.

- for calibration, it is probably not necessary to convert words
  back to integer id's. I understand that in C++ it is preferrable
  to use integers. On the other hand the calibration setup has human
  readable intermediate file formats, and there are only 2 sources
  of the 'words': the words.txt file and the ARPA LM.
  In case of mismatch, the script would simply fail with 'KeyError'.

chenguoguo committed:
Fixing compatibility issue with OS X in utils/prepare_lang.sh. Thanks Xavier Anguera for pointing this out.

danpovey committed:
Merge pull request #410 from chenguoguo/bugfix

Fixing compatibility issue with OS X in utils/prepare_lang.sh. Thanks…

KarelVesely84 committed:
calibration: adding comment on word filetering,

vijayaditya committed:
AMI: Minor correction to avoid overwrite of parallel alignments from different mics

danpovey committed:
Merge pull request #411 from vijayaditya/ami

AMI: Minor correction to avoid overwrite of parallel alignments from …

danpovey committed:
fix bug in word-align-lattice-lexicon.cc RE forced-out lattices.  Thanks to Xiaofei Yang for finding the issue.

danpovey committed:
changes to nnet3 version of PermuteComponent.  Not back-compatible (OK as AFAIK this component is not used yet).

vijayaditya committed:
Added results for AMI-IHM BLSTM

danpovey committed:
Changing error in wav-reading regarding wrong header to a warning; various other cosmetic changes

danpovey committed:
Merge pull request #418 from vijayaditya/new_bugifx

Added results for AMI-IHM BLSTM

danpovey committed:
nnet3: fix in LSTM training script to stop the final posterior-adjustment stage being very slow.

danpovey committed:
nnet3: minor changes to natural-gradient code to avoid NaNs appearing from zero data where minibatch size is very large; and to improve debuggability of that code.

danpovey committed:
change to tools/extras/check_dependencies.sh to give less misleading output on OpenSUSE where apt-get has been installed as part of a compatibility package.  Thanks to Colin Beckingham for pointing out the issue.

KarelVesely84 committed:
confidence: implementing next suggestions from Dan,

- replacing character-length of words by number of phonemes in a word,
- implementing word-specific feature file,
  - format: "word word-id filter  word-phone-length other_feats'
- replacing words with word-id's in 'steps/conf/prepare_calibration_data.py',

danpovey committed:
Merge pull request #385 from vesis84/confidence

Confidence calibration (ami, ihm)

tal1974 committed:
New option

New option for Using AffineComponentPreconditionOnline instead AffineComponent

tal1974 committed:
New option

Using AffineComponentPreconditionOnline instead AffineComponent

tal1974 committed:
New option

Using AffineComponentPreconditionOnline instead AffineComponent

tal1974 committed:
New option

Using AffineComponentPreconditionOnline instead AffineComponent

tal1974 committed:
New option

Using AffineComponentPreconditionOnline instead AffineComponent

danpovey committed:
Merge pull request #421 from tal1974/tal1974

New option for nnet1-to-raw-nnet

KarelVesely84 committed:
confidence: removing the log(#frames / #phones) feature as it was harmful

KarelVesely84 committed:
nnet1,cnn: bugfix of nnet-forward, reported by Ivan Kukanov. Thanks!

galv committed:
Fix expected opening tag of AffineComponent deserialization code.

danpovey committed:
Merge pull request #424 from galv/master

Fix expected opening tag of AffineComponent deserialization code.

vimalmanohar committed:
align_fix: Added --use-gpu option to nnet3-align-compiled

danpovey committed:
Merge pull request #425 from vimalmanohar/align_fix

align_fix: Added --use-gpu option to nnet3-align-compiled

freewym committed:
Add more results of BLSTM on swbd

danpovey committed:
Merge pull request #426 from freewym/master

Add more results of BLSTM on swbd

danpovey committed:
nnet3: adding more test for nnet components and their flags

xiaohui-zhang committed:
nnet2: applied changeds in the previous commit a92cb69 (addressing NaN issues) to nnet2

danpovey committed:
merging nnet-component-test.cc changes from chain branch, to accept slight differences from nondeterministic BLAS implementations

danpovey committed:
merging change from chain branch in matrix-lib-test.cc, testing for matrix library that is sensitive to NaNs between matrix rows

danpovey committed:
merging some changes to the natural gradient code from upstream/chain-- no difference in normal operation, they make it easier for debuggin.

xiaohui-zhang committed:
nnet2: applied changeds in the previous commits d240cf7 and a92cb69 (addressing NaN issues,.etc) to nnet2

freewym committed:
a minor bug in cu-sparse-matrix.cc

danpovey committed:
Merge pull request #429 from freewym/master

a minor bug in cu-sparse-matrix.cc

danpovey committed:
Revert "add bias init for nnet3"

danpovey committed:
Merge pull request #430 from kaldi-asr/revert-190-master

Revert "add bias init for nnet3"

danpovey committed:
bug fix in NaturalGradeantAffineComponent regarding how bias-mean option is interpreted

danpovey committed:
Merge pull request #428 from xiaohui-zhang/1509

nnet2: applied changeds in the previous commit a92cb69 (addressing Na…

danpovey committed:
bug fix in cu-sparse-matrix.cc regarding SparseMatrix with no elements.  Thanks to Mikolaj Pudo

danpovey committed:
bug fix in cu-sparse-matrix.cc regarding SparseMatrix with no elements.  Thanks to Mikolaj Pudo

chenguoguo committed:
Moving CuSparseMatrix/SparseMatrix out of CuVector/Vector

chenguoguo committed:
1. Changed CopyToVec in CuSparseMatrix/SparseMatrix to CopyElementsToVec, and removed OtherReal version. Note that I didn't remove the OtherReal version of CopyElementsToVec in SparseVector as that function is used by other function; 2. Replaced unnecessary CUDA kernels used in CopyElementsToVec, CopyColFromVec and CopyColFromMat with cublas_copy. Note, cuda_copy_col_from_mat_df and cuda_copy_col_from_mat_fd were not replaced.

chenguoguo committed:
Changing CuVector to CuVectorBase in CuSparseMatrix::CopyElementsToVec

maxhawkins committed:
const-correctness: make ComposeContext take a const reference

danpovey committed:
Merge pull request #436 from maxhawkins/master

const-correctness: make ComposeContext take a const reference

danpovey committed:
fix to previous commit regarding const-correctness: also change declaration.

danpovey committed:
Merge pull request #434 from chenguoguo/sparse_matrix

Moving CuSparseMatrix/SparseMatrix out of CuVector/Vector

pegahgh committed:
fixed inline const to enum in pitch-functions.* to avoid some compilation issue using visual studio

danpovey committed:
Merge pull request #437 from pegahgh/pegah-jan5

fixed inline const to enum in pitch-functions.* to avoid some compila…

danpovey committed:
minor documentation clarification

vijayaditya committed:
nnet3_to_dot.py : Added support for round descriptors and removed bug in sum descriptor processing

danpovey committed:
Merge pull request #439 from vijayaditya/new_feature

nnet3_to_dot.py : Added support for round descriptors and removed bug…

danpovey committed:
master: merging most code changes from chain branch (but not the actual chain code)-- want to keep shared parts of the code the same.

wintrode committed:
added flag to treat nnet call separately

wintrode committed:
Merge branch 'master' of https://github.com/wintrode/kaldi

wintrode committed:
Adding multilang recipe

danpovey committed:
documentation change: add entry to glossary

danpovey committed:
Add speed test for cudamatrix ApplyHeaviside that was missing

danpovey committed:
Modification to configure script to touch cudamatrix/cu-common.h when it is called- will tend to reduce compilation errors due to changing from not using to using CUDA, or vice versa.

Commit:
Tabs to spaces

There were some stray tabs in the Windows INSTALL file.

Commit:
Add note about patch failing due to line endings

Annoyingly, github's zip download doesn't seem to preserve line endings. See http://stackoverflow.com/questions/17347611/downloading-a-zip-from-github-removes-newlines-from-text-files

Commit:
Don't convert patch file line endings to LF

Patch on windows requires CRLF. This should leave them unchanged.

Commit:
Removed note about git archive mangling patch line endings

Should be fixed in the windows_line_endings branch.

Commit:
Don't mangle patch file line endings in all directories

Previous version would only look in the root dir which obviously doesn't work.

Commit:
Change line endings in Windows patch file to CRLF

The previous commits stop git mangling the file endings, but the file endings in the repository were LF, where they should have been CRLF.

danpovey committed:
modify extract_ivectors_online.sh to properly respect --online-ivector-period option (fix to issue found by remi francis)

danpovey committed:
fix to how CUDA block and grid sizes are computed for common operations, to work for larger matrices.

jtrmal committed:
Merge pull request #444 from Timmmm/windows_line_endings

Windows line endings

jtrmal committed:
Merge pull request #445 from Timmmm/windows_docs

Windows docs

Commit:
Clarify the generate_solution.pl command in the Windows INSTALL

It previously didn't mention that you need --enable-openblas (if using OpenBLAS).

Commit:
Formatting clean up and convert INSTALL to markdown

Commit:
Rename INSTALL to INSTALL.md so github renders it.

Timmmm committed:
Github markdown requires more spaces

danpovey committed:
minor documentation change

danpovey committed:
bug fix to a rather old script: get_lda_block.sh (in fact this reverts the most recent commit).

Commit:
Separated ARPA parsing from const LM construction

Commit:
Clarify that MLK and OpenBLAS are alternatives.

Commit:
Improved error messages for ARPA file parsing

arpa-file-parser.cc: Added a warning when declared count of n-grams is 0
const-arpa-lm.cc: Print an extended error message instead of simple ASSERT

Commit:
Changes per @danpovey's review in #458

danpovey committed:
Merge pull request #458 from kkm000/arpa-1

Separated ARPA parsing from const LM construction

jtrmal committed:
Merge pull request #448 from Timmmm/windows_docs

Windows docs

wintrode committed:
added generic multilang egs dir and config for coe grid

wintrode committed:
merged kaldi master with kaldi-wintrode (multilang eg)

wintrode committed:
babel multilang experiment

wintrode committed:
Merge branch 'master' of github.com:wintrode/kaldi
